omas/machine_mappings/d3d.py

```diff
@@ -31,255 +31,320 @@ def gas_injection_hardware(ods, pulse):
     if pulse < 100775:
         warnings.warn('DIII-D Gas valve locations not applicable for pulses earlier than 100775 (2000 JAN 17)')
 
-    i = 0
+    ip = 0
+    iv = 0
```

Comment:
Updating the `gas_injection_hardware` was heroic! Well done 🥇

Reply:
Thanks!  It was tedious, but pretty straightforward